server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1243,7 +1243,7 @@ private class ShardSnapshotsDeletion extends AbstractRunnable {
                 @Override
                 protected void doRun() throws Exception {
                     shardContainer = shardContainer(indexId, shardId);
-                    originalShardBlobs = shardContainer.listBlobs(OperationPurpose.SNAPSHOT).keySet();
+                    originalShardBlobs = shardContainer.listBlobs(OperationPurpose.SNAPSHOT_DATA).keySet();
```

Comment:
I've chosen `SNAPSHOT_DATA` here because most of the volume we're listing at this point will typically be data and not metadata, but really it could be either. It only matters for the write path. If we need more detailed purposes in future then we can add them.